gprofiler/main.py

```diff
@@ -289,26 +306,40 @@ def _snapshot(self):
         except Exception:
             logger.error("Running perf failed; consider running gProfiler with '--perf-mode none' to avoid using perf")
             raise
-
+        metadata = (
+            get_current_metadata(self._static_metadata)
+            if self._collect_metadata and self._client
+            else {"hostname": get_hostname()}
+        )
         if self._runtimes["system"]:
             merged_result, total_samples = merge.merge_profiles(
                 system_result,
                 process_profiles,
                 self._docker_client,
+                metadata,
```

Comment:
Why do we encode it with the profiles? This should be a different endpoint, what reason is there to mix them?

The existing (in master) `_make_profile_metadata` is a hack; it contains the container names because they really are required together with the profile itself, to do the mixing in the server.

All this metadata you added here - is not relevant (as far as I'm concerned?) so let's avoid mixing.

Reply:
Explained in https://github.com/Granulate/gprofiler/pull/127#discussion_r676918552

Reply:
We'll improve later...